api/src/main/java/com/mtech/vmcs/service/impl/CoinServiceImpl.java

```diff
@@ -35,4 +35,14 @@ public void updateCoins(List<Coin> coins) {
   public void deleteCoins(List<Long> coinIds) {
     coinRepository.deleteAllById(coinIds);
   }
+
+  @Override
+  public Boolean checkCoin(Coin coin){
+    for (Coin everyCoin:
+         getAllCoins()) {
+      if (everyCoin.getValue().equals(coin.getValue()))
```

Comment:
```checkCoin``` means validating the weight of coin, not value.

Reply:
Got it. I will change it later.